lib/proxy.js

```diff
@@ -110,9 +110,13 @@ exports.handler = function(req, res) {
       // don't save responses with codes other than 200
       if (!err && response.statusCode === 200) {
         var file = cache.write(dest);
-        r.pipe(file);
-        r.pipe(res, {end: false});
-
+        r.pipe(file).on('finish', function() {
+          cache.meta(dest, function(err, meta){
+            if (err)
+              throw err;
```

Comment:
It was bad idea to fix callbacks and exceptions. :( I should address it one day.